vxlan/vxlan.go

```diff
@@ -0,0 +1,277 @@
+// Package vxlan provides the capabilities to create a VXLAN overlay network
+package vxlan
+
+import (
+	"net"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	"github.com/rancher/go-rancher-metadata/metadata"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	changeCheckInterval = 5
+	metadataURL         = "http://rancher-metadata/2016-07-29"
+	vxlanMACRange       = "0E:00:00:00:00:00"
+	vxlanPort           = 4789
+)
+
+const (
+	DefaultBridgeName             = "docker0"
+	DefaultVxlanMTU               = 1500
```

Comment:
I have tested it. It can set the VTEP MTU to 9000.
But this seems to have caused some other problems,  I get 0 bandwidth by IPerf.

Maybe we should create a compatible jumbo frame issue.